### ISSUE TITLE
Terragrunt errors on empty terraform.tfstate files

### ISSUE THREAD

**Issue by silverlyra** [gruntwork-io/terragrunt#1785]
After migrating state from a `local` Terraform backend to a remote one, Terraform writes an empty `terraform.tfstate` file. This causes the next Terragrunt operation to fail with:

```
ERRO[0000] unexpected end of JSON input
```

After I figured out the empty `.tfstate` file was the culprit, working around this is easy, but I was scratching my head for awhile trying to figure out how to get Terragrunt working again.

**Comment by brikis98** [comment]
Huh, I wonder why Terraform writes that empty file? And what should Terragrunt's behavior here be? Ignore empty files? Find empty files, recognize them as an artifact of state migration, and keep looking for the "real" state file? Are there other cases where Terraform creates empty state files?

**Comment by silverlyra** [comment]
> Huh, I wonder why Terraform writes that empty file?

Yeah I don’t know. It might not be intentional behavior; maybe it was just more convenient to truncate the file than remove it, for some reason.

> And what should Terragrunt's behavior here be? Ignore empty files? Find empty files, recognize them as an artifact of state migration, and keep looking for the "real" state file?

In my original message, I see I didn’t make it clear that `unexpected end of JSON input` is the _only_ output from Terragrunt when this happens. As I remember, figuring out that the JSON file Terragrunt was trying to read was the empty `.tfstate` file took me more than an hour.

Adding something to the error message to indicate where the problem is would’ve made this a non-issue for me. I also think treating an empty file as if it did not exist (`rm`ing it was all it took) would be OK, since Terraform itself does create them and does subsequently ignore them. But if you want to be conservative and not change behavior, a better error message would work great.
